common/src/main/java/org/broadleafcommerce/common/resource/service/ResourceMinificationServiceImpl.java

```diff
@@ -64,20 +80,45 @@
     @Value("${minify.disableOptimizations}")
     protected boolean disableOptimizations;
 
+    @Value("${minify.closure.compiler.enabled}")
+    protected boolean closureCompilerEnabled;
+
+    protected CompilerOptions compilerOptions;
+
+    protected SystemPropertiesService systemPropertiesService;
+
+    @Autowired
+    @Qualifier("blClosureCompilerOptions")
+    public void setCompilerOptions(CompilerOptions compilerOptions) {
+        this.compilerOptions = compilerOptions;
+    }
+
+    @Autowired
+    @Qualifier("blSystemPropertiesService")
+    public void setSystemPropertiesService(SystemPropertiesService systemPropertiesService) {
+        this.systemPropertiesService = systemPropertiesService;
+    }
+
+    @Lookup
+    protected Compiler getClosureCompiler() {
+        return null;
+    }
+
```

Comment:
I'm guessing this is the reason on why the `Compiler` was a prototype bean. I'm not really sure what advantage making the `Compiler` bean a prototype is with using `@Lookup` instead of simply creating a singleton bean and injecting it is.